--- servo/components/style/build_gecko.rs.old	2018-12-15 18:28:45.115018252 +0000
+++ servo/components/style/build_gecko.rs	2018-12-15 18:29:12.463454795 +0000
@@ -139,7 +139,10 @@
 
             // Disable rust unions, because we replace some types inside of
             // them.
-            let mut builder = Builder::default().rust_target(RustTarget::Stable_1_0);
+            let mut builder = Builder::default().rust_target(RustTarget::Stable_1_0)
+                .clang_arg("-target")
+                .clang_arg("arm-linux-gnueabihf")
+                .clang_arg("-I/usr/arm-linux-gnueabihf/include/c++/4.8.4/arm-linux-gnueabihf");
 
             let rustfmt_path = env::var_os("RUSTFMT")
                 // This can be replaced with
