diff --git a/media/webrtc/trunk/webrtc/common_audio/common_audio_c_arm_asm_gn/moz.build b/media/webrtc/trunk/webrtc/common_audio/common_audio_c_arm_asm_gn/moz.build
index ca041b5..3ed0f78 100644
--- a/media/webrtc/trunk/webrtc/common_audio/common_audio_c_arm_asm_gn/moz.build
+++ b/media/webrtc/trunk/webrtc/common_audio/common_audio_c_arm_asm_gn/moz.build
@@ -132,6 +132,10 @@ if CONFIG["CPU_ARCH"] == "arm":
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
 
+    ASFLAGS += [
+        "-march=armv7-a"
+    ]
+
     SOURCES += [
         "/media/webrtc/trunk/webrtc/common_audio/signal_processing/complex_bit_reverse_arm.S",
         "/media/webrtc/trunk/webrtc/common_audio/signal_processing/filter_ar_fast_q12_armv7.S",
