--- media/webrtc/trunk/webrtc/common_audio/common_audio_c_gn/moz.build.old	2018-12-16 03:40:57.446902011 +0000
+++ media/webrtc/trunk/webrtc/common_audio/common_audio_c_gn/moz.build	2018-12-16 03:49:21.658580753 +0000
@@ -202,6 +202,10 @@
         "-mfpu=neon"
     ]
 
+    ASFLAGS += [
+        "-march=armv7-a"
+    ]
+
     DEFINES["WEBRTC_ARCH_ARM"] = True
     DEFINES["WEBRTC_ARCH_ARM_V7"] = True
     DEFINES["WEBRTC_HAS_NEON"] = True
